tool.ts

```diff
@@ -203,19 +203,19 @@ export async function downloadTool(url: string, fileName?:string): Promise<strin
             }
 
             // TODO: retries
+            tl.debug('downloading');
```

Comment:
@ericsciple @bryanmacfarlane  Could you check to ensure this makes sense? I moved the download code outside of the file open for two reasons:

1. There is a lingering file created regardless of the response of the HTTP request. If we get a non 200 request then I assume we don't want to create a file. This was causing lingering files (named as a UUID) to be created.
2. When adding the test for this code there were issues in the cleanup. I think the file stream was not closing on time so when the test code tried to clean up the TEMP directory it was running into issues. According to [this](https://github.com/shelljs/shelljs/issues/49) from shelljs there is a Windows issue here but they implemented a retry within a reasonable time limit. I am guessing our code goes beyond that.

Does this make sense/is there a better way to solve it?

Reply:
Is eric's question here addressed?  please ack

Reply:
@bryanmacfarlane : @ericsciple and I talked about this but I forget where we landed.

On error the stream is not closed ["Note: The stream is not closed when the 'error' event is emitted."](https://nodejs.org/api/stream.html#stream_event_error)

It looks like we can ensure the stream is closed by using [file.end()](https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback)

I am not sure if it's better to add it inside a finally, inside our .on('error'), or both.

Is there a standard to follow? From what I can infer it seems preferable to do it inside the latter.

Reply:
@scottdallamura Any advice here?